src/command_modules/azure-cli-datalake/azure/cli/command_modules/datalake/store/custom.py

```diff
@@ -0,0 +1,275 @@
+    ADLDownloader(
+        client,
+        source_path,
+        destination_path,
+        thread_count,
+        overwrite=overwrite)
+
+def test_adls_item(account_name,
+                   path):
+    return cf_datalake_store_filesystem(account_name).exists(path)
+
+# pylint: disable=redefined-variable-type
+def preview_adls_item(account_name,
+                      path,
+                      length=None,
+                      offset=0,
+                      force=False):
+    client = cf_datalake_store_filesystem(account_name)
+    if length:
+        try:
+            length = long(length)
+        except NameError:
+            length = int(length)
+
+    if offset:
+        try:
+            offset = long(offset)
+        except NameError:
+            offset = int(offset)
+
+    if not length or length <= 0:
+        length = client.info(path)['length'] - offset
+        if length > 1*1024*1024 and not force:
+            # pylint: disable=line-too-long
+            raise CLIError('The remaining data to preview is greater than {} bytes. Please specify a length or use the --force parameter to preview the entire file. The length of the file that would have been previewed: {}'.format(str(1*1024*1024), str(length)))
+
+    return client.read_block(path, offset, length)
+
+def join_adls_items(account_name,
+                    source_paths,
+                    destination_path,
+                    force=False):
+    client = cf_datalake_store_filesystem(account_name)
+    if force and client.exists(destination_path):
+        client.rm(destination_path)
+
+    client.concat(destination_path, source_paths)
+
+def move_adls_item(account_name,
+                   source_path,
+                   destination_path,
+                   force=False):
+    client = cf_datalake_store_filesystem(account_name)
+    if force and client.exists(destination_path):
+        client.rm(destination_path)
+    client.mv(source_path, destination_path)
+
+# pylint: disable=superfluous-parens
+def set_adls_item_expiry(account_name,
+                         path,
+                         expiration_time):
+    client = cf_datalake_store_filesystem(account_name)
+    if client.info(path)['type'] != 'FILE':
+        # pylint: disable=line-too-long
+        raise CLIError('The specified path does not exist or is not a file. Please ensure the path points to a file and it exists. Path supplied: {}'.format(path))
+    print('todo: {}'.format(expiration_time))
+
+# filesystem permissions customizations
+# pylint: disable=superfluous-parens
+def get_adls_item_acl():
+    print('todo')
+# pylint: disable=superfluous-parens
+def remove_adls_item_acl():
+    print('todo')
+# pylint: disable=superfluous-parens
+def remove_adls_item_acl_entry():
+    print('todo')
+# pylint: disable=superfluous-parens
+def set_adls_item_acl():
+    print('todo')
+# pylint: disable=superfluous-parens
+def set_adls_item_acl_entry():
+    print('todo')
```

Comment:
There are not exposed are they?  Instead of print('todo') I'd recommend raising an exception. `raise CLIError('Not implemented.')`  But most importantly make sure they aren't actually exposed. :)